sunflower-land/sunflower-land
# [FEAT] Faction Membership Updates

# Description

This PR includes the following functionality:

1. Leave a faction
2. Dynamic prices to join factions (5 SFL for bottom faction, 50 SFL for winning faction) - based on previous week.
3. No longer give emblems when joining a faction

It also cleans up the old pledging events.

<img width="536" alt="Screenshot 2024-07-10 at 7 23 15 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/c868fd56-8d09-4c3c-8421-7b2ef04516af">
<img width="522" alt="Screenshot 2024-07-10 at 7 23 22 PM" src="https://github.com/sunflower-land/sunflower-land/assets/11745561/11fbf1c6-1cea-4534-b0cf-d2a807e8878e">

**Leaving a faction**

So you're a traitor huh? To leave a faction, you must satisfy the following criteria:

1. No remaining emblems (you must sell them all)
2. You did not recently join (in last 24 hours).

When you leave a faction you will be punished and all marks will be lost as well as your banner.

## How to test?

1. Run against local API
2. Leave a faction (You can find the button at the bottom of the Emblem Board)
3. Join a faction

Ensure when leaving, that all marks and banners are removed.

The previous week winners, will determine the price to join a faction. You should walk around the Kingdom and notice that each recruiter is offering a different price to join - the Champions will be most expensive (50 SFL)

Note, there is a 24 hour cooldown after joining a faction before you can leave. You may need to comment this code out if you want to test multiple factions and/or update the DB pledgedAt value.
